### PR TITLE
Copy node between studies

### PR DESCRIPTION
**Pull request by AbdelHedhili**

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -166,18 +166,20 @@ public ResponseEntity<Void> deleteStudy(@PathVariable("studyUuid") UUID studyUui
         return ResponseEntity.ok().build();
     }
 
-    @PostMapping(value = "/studies/{studyUuid}/tree/nodes", params = {"nodeToCopyUuid", "referenceNodeUuid", "insertMode"})
+    @PostMapping(value = "/studies/{targetStudyUuid}/tree/nodes", params = {"nodeToCopyUuid", "referenceNodeUuid", "insertMode"})
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyController.java:169]: missing param sourceStudyUuid ?

File: src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java [hunk 2]
```diff
@@ -427,9 +427,9 @@ public String getSuffixedNodeName(UUID studyUuid, String nodeName) {
                 .map(AbstractNodeInfoEntity::getName)
                 .collect(Collectors.toList());
 
-        String uniqueName = StringUtils.EMPTY;
+        String uniqueName = nodeName;
         int i = 1;
-        while (StringUtils.EMPTY.equals(uniqueName) || studyNodeNames.contains(uniqueName)) {
+        while (studyNodeNames.contains(uniqueName)) {
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java:432]: why this change ?

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (jonenst) [reply src/main/java/org/gridsuite/study/server/StudyController.java:169]: not for optional params, only used to narrow down the mapping

**Review by jonenst — APPROVED** [review]
